packages/web/src/components/apps/statusbar.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import { BugIcon, ChevronDownIcon, ChevronRightIcon, TrashIcon } from 'lucide-react';
+import { useHotkeys } from 'react-hotkeys-hook';
+
+import { Button } from '@srcbook/components/src/components/ui/button';
+import { cn } from '@/lib/utils.ts';
+import { LogMessage, useLogs } from './use-logs';
+
+function getLabelForError(error: LogMessage) {
+  switch (error.type) {
+    case 'vite_error':
+      return 'Error running vite preview server!';
+    case 'npm_install_error':
+      return 'Error running npm install!';
+  }
+}
+
+type CollapsibleErrorMessageProps = {
+  error: LogMessage;
+};
+
+function CollapsibleErrorMessage({ error }: CollapsibleErrorMessageProps) {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      <button
+        className={cn(
+          'flex items-center gap-2 font-mono p-2 first:border-t first:border-t-border focus-visible:outline-none ring-inset focus-visible:ring-1 focus-visible:ring-ring',
+          {
+            'border-b border-b-border': !open,
+          },
+        )}
+        onClick={() => setOpen((n) => !n)}
+      >
+        {open ? <ChevronDownIcon size={16} /> : <ChevronRightIcon size={16} />}
+        <span className="text-tertiary-foreground select-none pointer-events-none">
+          {error.timestamp.toISOString()}{' '}
+        </span>
+        {getLabelForError(error)}
+      </button>
+      {open ? (
+        <pre className={cn('text-sm p-2', { 'ml-[15px] pl-4 mb-4 border-l': open })}>
+          {error.contents}
+        </pre>
+      ) : null}
+    </>
+  );
+}
+
+export default function Statusbar() {
+  const { logs, clearLogs, unreadLogsCount, open, togglePane } = useLogs();
+
+  useHotkeys('mod+shift+y', () => {
+    togglePane();
+  });
+
+  return (
+    <>
+      <div className="flex items-center justify-between h-8 border-t border-t-border px-2">
+        <Button
```

Comment:
Nit: logic here is a little harder to follow. Would it be more clear to use a:
```
{ open ? (
 .... // open tsx
) : (
 .... // closed tsx
)}

Reply:
What you've described is what I am doing in effect a little lower, though instead of no longer rendering a `div`, I'm adjusting its height so that it animates in:

```typescript
<div
  className={cn('flex flex-col overflow-auto cursor-text', 'h-0', { 'h-[320px]': open })}
>
  ...
</div>
```

I'm not sure there's much I can do to simplify the logic you are calling out - this stuff controls whether the "unread" badge is visible and whether the delete icon is visible. As originally conceived, those are a little more nuanced in when they are shown/hidden!